File: Problems/Blind75/ArrayHashing/contains_duplicate.rs
// https://leetcode.com/problems/contains-duplicate/
use std::collections::HashSet;

pub fn contains_duplicate(nums: Vec<i32>) -> bool {
    let mut numSet = HashSet::new();

    for n in nums {
        if !numSet.insert(n) { return true; }
    }

    false
}

File: Problems/Neetcode-150/arrays_hashing/contains-duplicate.cpp
// https://leetcode.com/problems/contains-duplicate/

class Solution {
public:
  bool containsDuplicate(vector<int> &nums) {
    unordered_set<int> numsSet;

    for (int num : nums) {
      if (numsSet.count(num))
        return true;

      numsSet.insert(num);
    }
    return false;
  }
};